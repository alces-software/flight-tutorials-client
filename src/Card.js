// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Software Ltd.
 *
 * This file is part of Flight Launch.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react';
import classNames from 'classnames';

import CardTitle from './CardTitle';

type PropTypes = {
  /* eslint-disable react/require-default-props */
  children : React$Element<*>,
  className? : string,
  onClick : () => void,
  footer? : React$Element<*>,
  subtitle? : string | React$Element<*>,
  subtitleSize? : 'medium' | 'large' | 'x-large',
  title : string | React$Element<*>,
  titleIcon? : React$Element<*>,
  titleLogoOnRight : boolean,
  titleLogoUrl? : string,
  titlePopoverText? : string,
  titleSize? : 'medium' | 'large' | 'x-large',
  /* eslint-enable react/require-default-props */
};

const Card = ({
  children,
  className,
  onClick,
  footer,
  subtitle,
  subtitleSize,
  title,
  titleIcon,
  titleLogoOnRight,
  titleLogoUrl,
  titlePopoverText,
  titleSize,
}: PropTypes) => {
  const cardClassNames = classNames('card', className, {
    'card--logo-right': titleLogoOnRight,
  });

  return (
    <div className="card-wrapper" >
      { /* eslint-disable jsx-a11y/no-static-element-interactions */ }
      <div className={cardClassNames} onClick={onClick} >
        <div className="card-block">
          <CardTitle
            title={title}
            titleSize={titleSize}
            titlePopoverText={titlePopoverText}
            subtitle={subtitle}
            subtitleSize={subtitleSize}
            logoUrl={titleLogoUrl}
            icon={titleIcon}
          />
          {children}
          {
            footer ? <div className="card-footer">
              {footer}
            </div>
            :
          null
          }
        </div>
      </div>
    </div>
  );
};


Card.Title = CardTitle;

export default Card;
