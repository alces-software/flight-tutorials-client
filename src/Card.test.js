// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';

import Card from './Card';

function renderCard(props) {
  return (
    <Card
      onClick={() => {}}
      subtitle="Card 1 subtitle"
      subtitleSize="medium"
      title="Card 1"
      titleLogoOnRight
      titleSize="large"
      {...props}
    >
      Card 1 description.
    </Card>
  );
};

it('renders without crashing', () => {
  const node = document.createElement('div');
  render(renderCard(), node);
});

it('renders correctly without footer', () => {
  const tree = renderer.create(
    renderCard()
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
it('renders correctly with footer', () => {
  const tree = renderer.create(
    renderCard({ footer: <div>My footer</div> })
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
