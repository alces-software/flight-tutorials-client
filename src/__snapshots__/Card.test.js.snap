// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with footer 1`] = `
<div
  className="card-wrapper"
>
  <div
    className="card card--logo-right"
    onClick={[Function]}
  >
    <div
      className="card-block"
    >
      <div
        className="card-title-block"
      >
        <div
          className="card-title-block-inner"
        >
          <div
            className="card-title-logo-container"
          >
            <div
              className="card-title-logo-helper"
            />
          </div>
          <h2
            className="card-title large"
            title="Card 1"
          >
            Card 1
          </h2>
        </div>
        <h6
          className="card-subtitle medium"
        >
          Card 1 subtitle
        </h6>
      </div>
      Card 1 description.
      <div
        className="card-footer"
      >
        <div>
          My footer
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly without footer 1`] = `
<div
  className="card-wrapper"
>
  <div
    className="card card--logo-right"
    onClick={[Function]}
  >
    <div
      className="card-block"
    >
      <div
        className="card-title-block"
      >
        <div
          className="card-title-block-inner"
        >
          <div
            className="card-title-logo-container"
          >
            <div
              className="card-title-logo-helper"
            />
          </div>
          <h2
            className="card-title large"
            title="Card 1"
          >
            Card 1
          </h2>
        </div>
        <h6
          className="card-subtitle medium"
        >
          Card 1 subtitle
        </h6>
      </div>
      Card 1 description.
    </div>
  </div>
</div>
`;
